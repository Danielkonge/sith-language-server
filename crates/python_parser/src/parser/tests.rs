mod context;
mod function;
mod invalid_assignments;
mod parser;
mod suite;
