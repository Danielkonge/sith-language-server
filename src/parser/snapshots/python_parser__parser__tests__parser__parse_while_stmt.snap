---
source: src/parser/tests.rs
expression: "parse(\"\nwhile x:\n    ...\nwhile (x > 1) and y:\n    pass\nelse:\n    'uwu'\nwhile x and y:\n    ...\n    print('Hello World!')\n\nelse:\n    print('Olá, Mundo!')\n    ...\nwhile a := b: ...\n\")"
---
ParsedFile {
    ast: Module {
        body: [
            While(
                WhileStmt {
                    test: Id(
                        Identifier {
                            id: "x",
                            range: 7..8,
                        },
                    ),
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 14..17,
                                    },
                                ),
                                range: 14..17,
                            },
                        ),
                    ],
                    orelse: [],
                    range: 1..17,
                },
            ),
            While(
                WhileStmt {
                    test: BoolOp(
                        BoolOpExpr {
                            op: And,
                            values: [
                                Compare(
                                    CompareExpr {
                                        left: Id(
                                            Identifier {
                                                id: "x",
                                                range: 25..26,
                                            },
                                        ),
                                        ops: [
                                            Gt,
                                        ],
                                        comparators: [
                                            Literal(
                                                LiteralExpr {
                                                    value: Int(
                                                        "1",
                                                    ),
                                                    range: 29..30,
                                                },
                                            ),
                                        ],
                                        range: 25..30,
                                    },
                                ),
                                Id(
                                    Identifier {
                                        id: "y",
                                        range: 36..37,
                                    },
                                ),
                            ],
                            range: 24..37,
                        },
                    ),
                    body: [
                        Pass(
                            PassStmt {
                                range: 43..47,
                            },
                        ),
                    ],
                    orelse: [
                        Expression(
                            ExpressionStmt {
                                value: Literal(
                                    LiteralExpr {
                                        value: String(
                                            "uwu",
                                        ),
                                        range: 58..63,
                                    },
                                ),
                                range: 58..63,
                            },
                        ),
                    ],
                    range: 18..63,
                },
            ),
            While(
                WhileStmt {
                    test: BoolOp(
                        BoolOpExpr {
                            op: And,
                            values: [
                                Id(
                                    Identifier {
                                        id: "x",
                                        range: 70..71,
                                    },
                                ),
                                Id(
                                    Identifier {
                                        id: "y",
                                        range: 76..77,
                                    },
                                ),
                            ],
                            range: 70..77,
                        },
                    ),
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 83..86,
                                    },
                                ),
                                range: 83..86,
                            },
                        ),
                        Expression(
                            ExpressionStmt {
                                value: Call(
                                    CallExpr {
                                        lhs: Id(
                                            Identifier {
                                                id: "print",
                                                range: 91..96,
                                            },
                                        ),
                                        args: Arguments {
                                            args: [
                                                Literal(
                                                    LiteralExpr {
                                                        value: String(
                                                            "Hello World!",
                                                        ),
                                                        range: 97..111,
                                                    },
                                                ),
                                            ],
                                            kw_args: [],
                                            range: 96..112,
                                        },
                                        range: 91..112,
                                    },
                                ),
                                range: 91..112,
                            },
                        ),
                    ],
                    orelse: [
                        Expression(
                            ExpressionStmt {
                                value: Call(
                                    CallExpr {
                                        lhs: Id(
                                            Identifier {
                                                id: "print",
                                                range: 124..129,
                                            },
                                        ),
                                        args: Arguments {
                                            args: [
                                                Literal(
                                                    LiteralExpr {
                                                        value: String(
                                                            "Olá, Mundo!",
                                                        ),
                                                        range: 130..144,
                                                    },
                                                ),
                                            ],
                                            kw_args: [],
                                            range: 129..145,
                                        },
                                        range: 124..145,
                                    },
                                ),
                                range: 124..145,
                            },
                        ),
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 150..153,
                                    },
                                ),
                                range: 150..153,
                            },
                        ),
                    ],
                    range: 64..153,
                },
            ),
            While(
                WhileStmt {
                    test: Named(
                        NamedExpr {
                            target: Id(
                                Identifier {
                                    id: "a",
                                    range: 160..161,
                                },
                            ),
                            value: Id(
                                Identifier {
                                    id: "b",
                                    range: 165..166,
                                },
                            ),
                            range: 160..166,
                        },
                    ),
                    body: [
                        Expression(
                            ExpressionStmt {
                                value: Ellipsis(
                                    EllipsisExpr {
                                        range: 168..171,
                                    },
                                ),
                                range: 168..171,
                            },
                        ),
                    ],
                    orelse: [],
                    range: 154..171,
                },
            ),
        ],
        range: 0..171,
    },
    parse_errors: [],
}
